pyeapi/api/staticroute.py

```diff
@@ -0,0 +1,302 @@
+# BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+# WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE
+# OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN
+# IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+"""Module for working with EOS static routes
+
+The staticroute resource provides configuration management of static
+route resources on an EOS node. It provides the following class
+implementations:
+
+    * StaticRoute - Configure static routes in EOS
+
+StaticRoute Attributes:
+    ip_dest (string): The ip address of the destination in the
+        form of A.B.C.D/E
+    next_hop (string): The next hop interface or ip address
+    next_hop_ip (string): The next hop address on destination interface
+    distance (int): Administrative distance for this route
+    tag (int): Route tag
+    route_name (string): Route name
+
+Notes:
+    The 'default' prefix function of the 'ip route' command,
+    'default ip route ...', currently equivalent to the 'no ip route ...'
+    command.
+"""
+
+import re
+
+from pyeapi.api import EntityCollection
+
+# Define the regex to match ip route lines (by lines in regex):
+#   'ip route' header
+#   ip_dest
+#   next_hop
+#   next_hop_ip
+#   distance
+#   tag
+#   name
+ROUTES_RE = re.compile(r'(?<=^ip route)'
+                       r' (\d+\.\d+\.\d+\.\d+\/\d+)'
+                       r' (\d+\.\d+\.\d+\.\d+|\S+)'
+                       r'(?: (\d+\.\d+\.\d+\.\d+))?'
+                       r' (\d+)'
+                       r'(?: tag (\d+))?'
+                       r'(?: name (\S+))?', re.M)
+
+# Define a format for the unique route id
+# The four parts in order are ip_dest, next_hop, next_hop_ip, and distance
+ROUTE_ID = "%s--%s--%s--%s"
+
+
+class StaticRoute(EntityCollection):
+    """The StaticRoute class provides a configuration instance
+    for working with static routes
+
+    """
+
+    def __str__(self):
+        return 'StaticRoute'
+
+    def get(self, ip_dest, next_hop, distance, next_hop_ip=None):
+        """Retrieves the ip route information for the route specified
+            by the ip_dest, next_hop, and distance parameters
+
+        Args:
+            ip_dest (string): The ip address of the destination in the
+                form of A.B.C.D/E
+            next_hop (string): The next hop interface or ip address
+            distance (int): Administrative distance for this route
+
+        Returns:
+            dict: An ip route dict object
+
+            If the unique route specified by the ip_dest, next_hop, and
+            distance does not exist, then None is returned.
+        """
+
+        # If distance is None, then set to 1 to match what EOS will
+        # do when distance is not specified.
+        if distance is None:
+            distance = 1
+
+        # Make the unique route_id for the requested route
+        route_id = ROUTE_ID % (ip_dest, next_hop, next_hop_ip, distance)
```

Comment:
FYI a side discussion on the EOS modules page is what to use as a field
separator when doing composite variables. I think we have decided to use
':' except between an IP address and the masklen where we follow convention
and use a slash (i.e. CIDR).

This may be only a puppet thing.

John

On Thu, Oct 15, 2015 at 1:36 PM, Philip DiLeo <notifications@github.com>
wrote:

> In pyeapi/api/staticroute.py
> <https://github.com/arista-eosplus/pyeapi/pull/45#discussion_r42176798>:
>
> > +            distance (int): Administrative distance for this route
> > +
> > +        Returns:
> > +            dict: An ip route dict object
> > +
> > +            If the unique route specified by the ip_dest, next_hop, and
> > +            distance does not exist, then None is returned.
> > +        """
> > +
> > +        # If distance is None, then set to 1 to match what EOS will
> > +        # do when distance is not specified.
> > +        if distance is None:
> > +            distance = 1
> > +
> > +        # Make the unique route_id for the requested route
> > +        route_id = ROUTE_ID % (ip_dest, next_hop, next_hop_ip, distance)
>
> You can just pass the vars into line 115. No need to create a new variable.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/arista-eosplus/pyeapi/pull/45/files#r42176798>.
>